app/controllers/registrations_controller.rb

```diff
@@ -5,14 +5,9 @@ def after_sign_up_path_for(resource)
     # add session information for tags
     session[:new_user_registration] = true;
 
-    # normal after_sign_up_path_for
     return_to = session[:return_to]
-    if return_to and return_to.include?('contributions')
-      return_to = session[:return_to]
-      session[:return_to] = nil
-      return return_to
-    else
-      return "/#{I18n.locale}/sign_up_success"
-    end
+    session[:return_to] = nil
+
+    return_to || sign_up_success_path
   end
```

Comment:
acho que o Devise tem algo pra isso, da uma olhada
https://github.com/plataformatec/devise/wiki/How-To:-Redirect-back-to-current-page-after-sign-in,-sign-out,-sign-up,-update